modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -170,11 +171,19 @@ def create_pod
       print_good("Pod created: #{pod_name}")
 
       print_status('Waiting for the pod to be ready...')
-      10.times do
-        sleep 3
+      ready = retry_until_true(timeout: datastore['PodReadyTimeout']) do
         pod = @kubernetes_client.get_pod(pod_name, namespace)
-        ready = pod.dig(:status, :containerStatuses).find { |status| status[:name] = @pod_name }[:ready]
-        break if ready
+        pod_status = pod[:status]
+        next if pod_status == 'Failure'
+
+        container_statuses = pod_status[:containerStatuses]
+        next unless container_statuses
+
+        ready = container_statuses.any? { |status| status[:name] = @pod_name && status[:ready] }
```

Comment:
I realize this issue predates your changes, but I think this should be the following right?

```suggestion
        ready = container_statuses.any? { |status| status[:name] == @pod_name && status[:ready] }
```

Reply:
Nevermind, it looks like the `status[:name] = @pod_name` expression was never relevant here. The name is actually the name of the container, not the pod so changing it to `==` breaks it.